### ISSUE TITLE
Issue: Mosquitto broker refusing connections, Anonymus mode not working

### ISSUE THREAD

**Issue by MatejsLab** [eclipse/mosquitto#2641]
I have Raspberry Pi 4, today I installed the latest Raspberry Pi OS 32-bit, and Mosquitto 2.0.11-1.

I installed mosquito via the command line, and now I can't connect to the broker with the default configuration(nothing changed, just installed).

```
systemctl status mosquitto.service
● mosquitto.service - Mosquitto MQTT Broker
     Loaded: loaded (/lib/systemd/system/mosquitto.service; enabled; vendor preset: enabled)
     Active: active (running) since Sat 2022-10-01 15:19:54 CEST; 4s ago
       Docs: man:mosquitto.conf(5)
             man:mosquitto(8)
    Process: 4260 ExecStartPre=/bin/mkdir -m 740 -p /var/log/mosquitto (code=exited, status=0/SUCCESS)
    Process: 4261 ExecStartPre=/bin/chown mosquitto /var/log/mosquitto (code=exited, status=0/SUCCESS)
    Process: 4262 ExecStartPre=/bin/mkdir -m 740 -p /run/mosquitto (code=exited, status=0/SUCCESS)
    Process: 4263 ExecStartPre=/bin/chown mosquitto /run/mosquitto (code=exited, status=0/SUCCESS)
   Main PID: 4264 (mosquitto)
      Tasks: 1 (limit: 3720)
        CPU: 61ms
     CGroup: /system.slice/mosquitto.service
             └─4264 /usr/sbin/mosquitto -c /etc/mosquitto/mosquitto.conf

Oct 01 15:19:54 pi3-dule systemd[1]: Starting Mosquitto MQTT Broker...
Oct 01 15:19:54 pi3-dule systemd[1]: Started Mosquitto MQTT Broker.

```

```
sudo netstat -tlnp | grep mosq
tcp        0      0 127.0.0.1:1883          0.0.0.0:*               LISTEN      4264/mosquitto      
tcp6       0      0 ::1:1883                :::*                    LISTEN      4264/mosquitto      

```
It looks like it works but it refuses to connect in anonymous mode.
I also tried allowing anonymous mode in conf.d on the image below, but no success.
 
![image](https://user-images.githubusercontent.com/28715097/193413430-341f4b48-f585-436e-adac-7b0dec01f073.png)

A year ago when I did the Mosquitto install, it just worked out of the gate, now I can't make it work. I searched on google and no one talks about that issue. I can't downgrade Mosquitto with the default repo as it says  2.0.11-1 is the only stable version.

I have had mosquito as a broker for Openhab for many years, last week the raspberry crashed and I can't restore it with timeshift (aparentlly timshift misbehaves as i have read), I also have 1-year-old sd card image, but I decided to do a fresh install and broken mosquito is preventing me from getting things back up and running.

Any help appriciated, like how i could install any other working version of mosquitto.

**Comment by MatejsLab** [comment]
That is the note for anyone having the same problem, that requires a quick fix.
The issue is not yet solved but I manage to bodge the older mosquitto version with snap.
I have read that mosquito 2.0 requires config to set up authentication, but I can't do that on 20 clients immediately.
So version 1. x by default accepts anonymous.

I used snap packet manager to install mosquitto 1.6.

sudo apt install snapd
sudo snap install mosquitto --channel=1.6/stable
sudo snap start mosquitto

Now it just works, as it did before.

But I think it is important that there would be a way to manually set **anonymus_mode true** without messing with config files.
Just for getting legacy systems up and running easier, as i am not keen on config files, and not a Linux pro.

**Comment by GravityRZ** [comment]
this still does not seem to be working in 2.0.15

i put 
listener 1883
allow_anonymous true

in the config file but it refuses to load.

man i am running toke mosquitto for ages without problems
every time i try to switch to eclipse it is just not working.

i download the image, create a docker container with the rifght settings(mapped to mosquitto/data  mosquitto/log  mosquitto/config

i see that it is touching the file but it refuses to load the files and refuses to stasrt also (address in use)

did it again with port 1884 and still the message address in use.
ofcourse the address is in use. it runs docker with multiple containers but non of theme are using 1884 as a port

**Comment by MatejsLab** [comment]
yeah, v2.0 still seems to be rubbish for legacy use with anonymous clients.
Try v1.6 for now, you can obtain it via snap as I did above.

**Comment by GravityRZ** [comment]
got 2.0.15 running stable

here is the mosquitto.conf i used so everybody can use it properly

```
log_timestamp true
log_timestamp_format %d-%m-%YT%H:%M:%S
persistence true
persistence_location /mosquitto/data
log_dest file /mosquitto/log/mosquitto.log
#password_file /mosquitto/config/pwfile
allow_anonymous true
listener 1883 192.168.1.50
```

this is the command i used to create and start it
please note i run it on a synology so you need to change the /volume1/docker path to tthe one you use

```
sudo docker run --name=EclipseMQTT -itd --restart=always --net=host -e TZ=Europe/Amsterdam
-v /volume1/docker/mqtt/config:/mosquitto/config -v /volume1/docker/mqtt/data:/mosquitto/data -v /volume1/docker/mqtt/log:/mosquitto/log -v /volume1/docker/mqtt/config/mosquitto.conf:/mosquitto/config/mosquitto.conf eclipse-mosquitto:openssl
```

**Comment by MatejsLab** [comment]
I haven't tried it yet, but thanks a lot for posting a configuration, that works for you, in the new mosquito.
